[BUG] test_dpp_reuse_broadcast_exchange failed

From a recent nightly test run:
```
[2024-01-03T17:41:18.525Z] FAILED ../../src/main/python/dpp_test.py::test_dpp_reuse_broadcast_exchange[true-5-parquet][DATAGEN_SEED=1704297021, INJECT_OOM, IGNORE_ORDER] - py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback.assertContains.
[2024-01-03T17:41:18.525Z] : java.lang.AssertionError: assertion failed: Could not find DynamicPruningExpression in the Spark plan
[2024-01-03T17:41:18.525Z] AdaptiveSparkPlan isFinalPlan=true
[2024-01-03T17:41:18.525Z] +- == Final Plan ==
[2024-01-03T17:41:18.525Z]    LocalTableScan <empty>, [key#37135, max(value)#37151L]
[2024-01-03T17:41:18.525Z] +- == Initial Plan ==
[2024-01-03T17:41:18.525Z]    Sort [key#37135 ASC NULLS FIRST, max(value)#37151L ASC NULLS FIRST], true, 0
[2024-01-03T17:41:18.525Z]    +- Exchange rangepartitioning(key#37135 ASC NULLS FIRST, max(value)#37151L ASC NULLS FIRST, 4), ENSURE_REQUIREMENTS, [plan_id=109491]
[2024-01-03T17:41:18.525Z]       +- HashAggregate(keys=[key#37135], functions=[max(value#37136L)], output=[key#37135, max(value)#37151L])
[2024-01-03T17:41:18.525Z]          +- Exchange hashpartitioning(key#37135, 4), ENSURE_REQUIREMENTS, [plan_id=109488]
[2024-01-03T17:41:18.525Z]             +- HashAggregate(keys=[key#37135], functions=[partial_max(value#37136L)], output=[key#37135, max#37199L])
[2024-01-03T17:41:18.525Z]                +- Union
[2024-01-03T17:41:18.525Z]                   :- Project [key#37054 AS key#37135, value#37140L AS value#37136L]
[2024-01-03T17:41:18.525Z]                   :  +- BroadcastHashJoin [key#37054], [key#37056], Inner, BuildRight, false
[2024-01-03T17:41:18.526Z]                   :     :- HashAggregate(keys=[key#37054], functions=[sum(value#37053)], output=[key#37054, value#37140L])
[2024-01-03T17:41:18.526Z]                   :     :  +- Exchange hashpartitioning(key#37054, 4), ENSURE_REQUIREMENTS, [plan_id=109473]
[2024-01-03T17:41:18.526Z]                   :     :     +- HashAggregate(keys=[key#37054], functions=[partial_sum(value#37053)], output=[key#37054, sum#37201L])
[2024-01-03T17:41:18.526Z]                   :     :        +- Project [value#37053, key#37054]
[2024-01-03T17:41:18.526Z]                   :     :           +- Filter (isnotnull(value#37053) AND (value#37053 > 0))
[2024-01-03T17:41:18.526Z]                   :     :              +- FileScan parquet spark_catalog.default.tmp_table_gw1_470723338_0[value#37053,key#37054,skey#37055] Batched: true, DataFilters: [isnotnull(value#37053), (value#37053 > 0)], Format: Parquet, Location: InMemoryFileIndex(50 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-git..., PartitionFilters: [isnotnull(key#37054), dynamicpruningexpression(key#37054 IN dynamicpruning#37196)], PushedFilters: [IsNotNull(value), GreaterThan(value,0)], ReadSchema: struct<value:int>
[2024-01-03T17:41:18.526Z]                   :     :                    +- SubqueryAdaptiveBroadcast dynamicpruning#37196, 0, true, Project [key#37056], [key#37056]
[2024-01-03T17:41:18.526Z]                   :     :                       +- AdaptiveSparkPlan isFinalPlan=false
[2024-01-03T17:41:18.526Z]                   :     :                          +- Project [key#37056]
[2024-01-03T17:41:18.526Z]                   :     :                             +- Filter ((((isnotnull(ex_key#37058) AND isnotnull(filter#37060)) AND (ex_key#37058 = 3)) AND (filter#37060 = 1458)) AND isnotnull(key#37056))
[2024-01-03T17:41:18.526Z]                   :     :                                +- FileScan parquet spark_catalog.default.tmp_table_gw1_470723338_1[key#37056,ex_key#37058,filter#37060] Batched: true, DataFilters: [isnotnull(ex_key#37058), isnotnull(filter#37060), (ex_key#37058 = 3), (filter#37060 = 1458), isn..., Format: Parquet, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.526Z]                   :     +- BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, true] as bigint)),false), [plan_id=109476]
[2024-01-03T17:41:18.526Z]                   :        +- Project [key#37056]
[2024-01-03T17:41:18.526Z]                   :           +- Filter ((((isnotnull(ex_key#37058) AND isnotnull(filter#37060)) AND (ex_key#37058 = 3)) AND (filter#37060 = 1458)) AND isnotnull(key#37056))
[2024-01-03T17:41:18.526Z]                   :              +- FileScan parquet spark_catalog.default.tmp_table_gw1_470723338_1[key#37056,ex_key#37058,filter#37060] Batched: true, DataFilters: [isnotnull(ex_key#37058), isnotnull(filter#37060), (ex_key#37058 = 3), (filter#37060 = 1458), isn..., Format: Parquet, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.526Z]                   +- Project [key#37184, value#37187L]
[2024-01-03T17:41:18.526Z]                      +- BroadcastHashJoin [key#37184], [key#37188], Inner, BuildRight, false
[2024-01-03T17:41:18.526Z]                         :- HashAggregate(keys=[key#37184], functions=[sum(value#37183)], output=[key#37184, value#37187L])
[2024-01-03T17:41:18.526Z]                         :  +- Exchange hashpartitioning(key#37184, 4), ENSURE_REQUIREMENTS, [plan_id=109479]
[2024-01-03T17:41:18.526Z]                         :     +- HashAggregate(keys=[key#37184], functions=[partial_sum(value#37183)], output=[key#37184, sum#37203L])
[2024-01-03T17:41:18.526Z]                         :        +- Project [value#37183, key#37184]
[2024-01-03T17:41:18.526Z]                         :           +- Filter (isnotnull(value#37183) AND (value#37183 > 0))
[2024-01-03T17:41:18.526Z]                         :              +- FileScan parquet spark_catalog.default.tmp_table_gw1_470723338_0[value#37183,key#37184,skey#37185] Batched: true, DataFilters: [isnotnull(value#37183), (value#37183 > 0)], Format: Parquet, Location: InMemoryFileIndex(50 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-git..., PartitionFilters: [isnotnull(key#37184), dynamicpruningexpression(key#37184 IN dynamicpruning#37197)], PushedFilters: [IsNotNull(value), GreaterThan(value,0)], ReadSchema: struct<value:int>
[2024-01-03T17:41:18.526Z]                         :                    +- SubqueryAdaptiveBroadcast dynamicpruning#37197, 0, true, Project [key#37188], [key#37188]
[2024-01-03T17:41:18.526Z]                         :                       +- AdaptiveSparkPlan isFinalPlan=false
[2024-01-03T17:41:18.526Z]                         :                          +- Project [key#37188]
[2024-01-03T17:41:18.526Z]                         :                             +- Filter ((((isnotnull(ex_key#37190) AND isnotnull(filter#37192)) AND (ex_key#37190 = 3)) AND (filter#37192 = 1458)) AND isnotnull(key#37188))
[2024-01-03T17:41:18.526Z]                         :                                +- FileScan parquet spark_catalog.default.tmp_table_gw1_470723338_1[key#37188,ex_key#37190,filter#37192] Batched: true, DataFilters: [isnotnull(ex_key#37190), isnotnull(filter#37192), (ex_key#37190 = 3), (filter#37192 = 1458), isn..., Format: Parquet, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.526Z]                         +- BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, true] as bigint)),false), [plan_id=109482]
[2024-01-03T17:41:18.526Z]                            +- Project [key#37188]
[2024-01-03T17:41:18.526Z]                               +- Filter ((((isnotnull(ex_key#37190) AND isnotnull(filter#37192)) AND (ex_key#37190 = 3)) AND (filter#37192 = 1458)) AND isnotnull(key#37188))
[2024-01-03T17:41:18.526Z]                                  +- FileScan parquet spark_catalog.default.tmp_table_gw1_470723338_1[key#37188,ex_key#37190,filter#37192] Batched: true, DataFilters: [isnotnull(ex_key#37190), isnotnull(filter#37192), (ex_key#37190 = 3), (filter#37192 = 1458), isn..., Format: Parquet, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.526Z] 
[2024-01-03T17:41:18.526Z] 	at scala.Predef$.assert(Predef.scala:223)
[2024-01-03T17:41:18.526Z] 	at org.apache.spark.sql.rapids.ShimmedExecutionPlanCaptureCallbackImpl.assertContains(ShimmedExecutionPlanCaptureCallbackImpl.scala:170)
[2024-01-03T17:41:18.526Z] 	at org.apache.spark.sql.rapids.ShimmedExecutionPlanCaptureCallbackImpl.assertContains(ShimmedExecutionPlanCaptureCallbackImpl.scala:175)
[2024-01-03T17:41:18.526Z] 	at org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback$.assertContains(ExecutionPlanCaptureCallback.scala:76)
[2024-01-03T17:41:18.526Z] 	at org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback.assertContains(ExecutionPlanCaptureCallback.scala)
[2024-01-03T17:41:18.526Z] 	at sun.reflect.GeneratedMethodAccessor140.invoke(Unknown Source)
[2024-01-03T17:41:18.526Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-01-03T17:41:18.526Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2024-01-03T17:41:18.526Z] 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-01-03T17:41:18.526Z] 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-01-03T17:41:18.526Z] 	at py4j.Gateway.invoke(Gateway.java:282)
[2024-01-03T17:41:18.526Z] 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-01-03T17:41:18.526Z] 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-01-03T17:41:18.526Z] 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-01-03T17:41:18.526Z] 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-01-03T17:41:18.527Z] 	at java.lang.Thread.run(Thread.java:750)
[2024-01-03T17:41:18.527Z] FAILED ../../src/main/python/dpp_test.py::test_dpp_reuse_broadcast_exchange[true-5-orc][DATAGEN_SEED=1704297021, INJECT_OOM, IGNORE_ORDER] - py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback.assertContains.
[2024-01-03T17:41:18.527Z] : java.lang.AssertionError: assertion failed: Could not find DynamicPruningExpression in the Spark plan
[2024-01-03T17:41:18.527Z] AdaptiveSparkPlan isFinalPlan=true
[2024-01-03T17:41:18.527Z] +- == Final Plan ==
[2024-01-03T17:41:18.527Z]    LocalTableScan <empty>, [key#37628, max(value)#37644L]
[2024-01-03T17:41:18.527Z] +- == Initial Plan ==
[2024-01-03T17:41:18.527Z]    Sort [key#37628 ASC NULLS FIRST, max(value)#37644L ASC NULLS FIRST], true, 0
[2024-01-03T17:41:18.527Z]    +- Exchange rangepartitioning(key#37628 ASC NULLS FIRST, max(value)#37644L ASC NULLS FIRST, 4), ENSURE_REQUIREMENTS, [plan_id=111165]
[2024-01-03T17:41:18.527Z]       +- HashAggregate(keys=[key#37628], functions=[max(value#37629L)], output=[key#37628, max(value)#37644L])
[2024-01-03T17:41:18.527Z]          +- Exchange hashpartitioning(key#37628, 4), ENSURE_REQUIREMENTS, [plan_id=111162]
[2024-01-03T17:41:18.527Z]             +- HashAggregate(keys=[key#37628], functions=[partial_max(value#37629L)], output=[key#37628, max#37692L])
[2024-01-03T17:41:18.527Z]                +- Union
[2024-01-03T17:41:18.527Z]                   :- Project [key#37547 AS key#37628, value#37633L AS value#37629L]
[2024-01-03T17:41:18.527Z]                   :  +- BroadcastHashJoin [key#37547], [key#37549], Inner, BuildRight, false
[2024-01-03T17:41:18.527Z]                   :     :- HashAggregate(keys=[key#37547], functions=[sum(value#37546)], output=[key#37547, value#37633L])
[2024-01-03T17:41:18.527Z]                   :     :  +- Exchange hashpartitioning(key#37547, 4), ENSURE_REQUIREMENTS, [plan_id=111147]
[2024-01-03T17:41:18.527Z]                   :     :     +- HashAggregate(keys=[key#37547], functions=[partial_sum(value#37546)], output=[key#37547, sum#37694L])
[2024-01-03T17:41:18.527Z]                   :     :        +- Project [value#37546, key#37547]
[2024-01-03T17:41:18.527Z]                   :     :           +- Filter (isnotnull(value#37546) AND (value#37546 > 0))
[2024-01-03T17:41:18.527Z]                   :     :              +- FileScan orc spark_catalog.default.tmp_table_gw1_1272856583_0[value#37546,key#37547,skey#37548] Batched: true, DataFilters: [isnotnull(value#37546), (value#37546 > 0)], Format: ORC, Location: InMemoryFileIndex(50 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-git..., PartitionFilters: [isnotnull(key#37547), dynamicpruningexpression(key#37547 IN dynamicpruning#37689)], PushedFilters: [IsNotNull(value), GreaterThan(value,0)], ReadSchema: struct<value:int>
[2024-01-03T17:41:18.527Z]                   :     :                    +- SubqueryAdaptiveBroadcast dynamicpruning#37689, 0, true, Project [key#37549], [key#37549]
[2024-01-03T17:41:18.527Z]                   :     :                       +- AdaptiveSparkPlan isFinalPlan=false
[2024-01-03T17:41:18.527Z]                   :     :                          +- Project [key#37549]
[2024-01-03T17:41:18.527Z]                   :     :                             +- Filter ((((isnotnull(ex_key#37551) AND isnotnull(filter#37553)) AND (ex_key#37551 = 3)) AND (filter#37553 = 1458)) AND isnotnull(key#37549))
[2024-01-03T17:41:18.527Z]                   :     :                                +- FileScan orc spark_catalog.default.tmp_table_gw1_1272856583_1[key#37549,ex_key#37551,filter#37553] Batched: true, DataFilters: [isnotnull(ex_key#37551), isnotnull(filter#37553), (ex_key#37551 = 3), (filter#37553 = 1458), isn..., Format: ORC, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.527Z]                   :     +- BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, true] as bigint)),false), [plan_id=111150]
[2024-01-03T17:41:18.527Z]                   :        +- Project [key#37549]
[2024-01-03T17:41:18.527Z]                   :           +- Filter ((((isnotnull(ex_key#37551) AND isnotnull(filter#37553)) AND (ex_key#37551 = 3)) AND (filter#37553 = 1458)) AND isnotnull(key#37549))
[2024-01-03T17:41:18.527Z]                   :              +- FileScan orc spark_catalog.default.tmp_table_gw1_1272856583_1[key#37549,ex_key#37551,filter#37553] Batched: true, DataFilters: [isnotnull(ex_key#37551), isnotnull(filter#37553), (ex_key#37551 = 3), (filter#37553 = 1458), isn..., Format: ORC, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.527Z]                   +- Project [key#37677, value#37680L]
[2024-01-03T17:41:18.527Z]                      +- BroadcastHashJoin [key#37677], [key#37681], Inner, BuildRight, false
[2024-01-03T17:41:18.527Z]                         :- HashAggregate(keys=[key#37677], functions=[sum(value#37676)], output=[key#37677, value#37680L])
[2024-01-03T17:41:18.527Z]                         :  +- Exchange hashpartitioning(key#37677, 4), ENSURE_REQUIREMENTS, [plan_id=111153]
[2024-01-03T17:41:18.527Z]                         :     +- HashAggregate(keys=[key#37677], functions=[partial_sum(value#37676)], output=[key#37677, sum#37696L])
[2024-01-03T17:41:18.527Z]                         :        +- Project [value#37676, key#37677]
[2024-01-03T17:41:18.527Z]                         :           +- Filter (isnotnull(value#37676) AND (value#37676 > 0))
[2024-01-03T17:41:18.527Z]                         :              +- FileScan orc spark_catalog.default.tmp_table_gw1_1272856583_0[value#37676,key#37677,skey#37678] Batched: true, DataFilters: [isnotnull(value#37676), (value#37676 > 0)], Format: ORC, Location: InMemoryFileIndex(50 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-git..., PartitionFilters: [isnotnull(key#37677), dynamicpruningexpression(key#37677 IN dynamicpruning#37690)], PushedFilters: [IsNotNull(value), GreaterThan(value,0)], ReadSchema: struct<value:int>
[2024-01-03T17:41:18.527Z]                         :                    +- SubqueryAdaptiveBroadcast dynamicpruning#37690, 0, true, Project [key#37681], [key#37681]
[2024-01-03T17:41:18.527Z]                         :                       +- AdaptiveSparkPlan isFinalPlan=false
[2024-01-03T17:41:18.527Z]                         :                          +- Project [key#37681]
[2024-01-03T17:41:18.527Z]                         :                             +- Filter ((((isnotnull(ex_key#37683) AND isnotnull(filter#37685)) AND (ex_key#37683 = 3)) AND (filter#37685 = 1458)) AND isnotnull(key#37681))
[2024-01-03T17:41:18.527Z]                         :                                +- FileScan orc spark_catalog.default.tmp_table_gw1_1272856583_1[key#37681,ex_key#37683,filter#37685] Batched: true, DataFilters: [isnotnull(ex_key#37683), isnotnull(filter#37685), (ex_key#37683 = 3), (filter#37685 = 1458), isn..., Format: ORC, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.527Z]                         +- BroadcastExchange HashedRelationBroadcastMode(List(cast(input[0, int, true] as bigint)),false), [plan_id=111156]
[2024-01-03T17:41:18.527Z]                            +- Project [key#37681]
[2024-01-03T17:41:18.527Z]                               +- Filter ((((isnotnull(ex_key#37683) AND isnotnull(filter#37685)) AND (ex_key#37683 = 3)) AND (filter#37685 = 1458)) AND isnotnull(key#37681))
[2024-01-03T17:41:18.527Z]                                  +- FileScan orc spark_catalog.default.tmp_table_gw1_1272856583_1[key#37681,ex_key#37683,filter#37685] Batched: true, DataFilters: [isnotnull(ex_key#37683), isnotnull(filter#37685), (ex_key#37683 = 3), (filter#37685 = 1458), isn..., Format: ORC, Location: InMemoryFileIndex(1 paths)[file:/home/jenkins/agent/workspace/jenkins-rapids_integration-dev-gith..., PartitionFilters: [], PushedFilters: [IsNotNull(ex_key), IsNotNull(filter), EqualTo(ex_key,3), EqualTo(filter,1458), IsNotNull(key)], ReadSchema: struct<key:int,ex_key:int,filter:int>
[2024-01-03T17:41:18.528Z] 
[2024-01-03T17:41:18.528Z] 	at scala.Predef$.assert(Predef.scala:223)
[2024-01-03T17:41:18.528Z] 	at org.apache.spark.sql.rapids.ShimmedExecutionPlanCaptureCallbackImpl.assertContains(ShimmedExecutionPlanCaptureCallbackImpl.scala:170)
[2024-01-03T17:41:18.528Z] 	at org.apache.spark.sql.rapids.ShimmedExecutionPlanCaptureCallbackImpl.assertContains(ShimmedExecutionPlanCaptureCallbackImpl.scala:175)
[2024-01-03T17:41:18.528Z] 	at org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback$.assertContains(ExecutionPlanCaptureCallback.scala:76)
[2024-01-03T17:41:18.528Z] 	at org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback.assertContains(ExecutionPlanCaptureCallback.scala)
[2024-01-03T17:41:18.528Z] 	at sun.reflect.GeneratedMethodAccessor140.invoke(Unknown Source)
[2024-01-03T17:41:18.528Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-01-03T17:41:18.528Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2024-01-03T17:41:18.528Z] 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-01-03T17:41:18.528Z] 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-01-03T17:41:18.528Z] 	at py4j.Gateway.invoke(Gateway.java:282)
[2024-01-03T17:41:18.528Z] 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-01-03T17:41:18.528Z] 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-01-03T17:41:18.528Z] 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-01-03T17:41:18.528Z] 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-01-03T17:41:18.528Z] 	at java.lang.Thread.run(Thread.java:750)
[2024-01-03T17:41:18.528Z] = 2 failed, 20760 passed, 933 skipped, 417 xfailed, 388 xpassed, 9504 warnings in 6656.39s (1:50:56) =
```

Basically by the plan output here, it looks like this is an AQE optimization that is turning the entire plan into a `LocalTableScan`

```
E                   py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback.assertContains.
E                   : java.lang.AssertionError: assertion failed: Could not find DynamicPruningExpression in the Spark plan
E                   AdaptiveSparkPlan isFinalPlan=true
E                   +- == Final Plan ==
E                      LocalTableScan <empty>, [key#2006, max(value)#2017L]
E                   +- == Initial Plan ==
E                   +- == Initial Plan ==
E                      Sort [key#2006 ASC NULLS FIRST, max(value)#2017L ASC NULLS FIRST], true, 0
E                      +- Exchange rangepartitioning(key#2006 ASC NULLS FIRST, max(value)#2017L ASC NULLS FIRST, 4), ENSURE_REQUIREMENTS, [plan_id=5302]
E                         +- HashAggregate(keys=[key#2006], functions=[max(value#2007L)], output=[key#2006, max(value)#2017L])
E                            +- Exchange hashpartitioning(key#2006, 4), ENSURE_REQUIREMENTS, [plan_id=5299]
E                               +- HashAggregate(keys=[key#2006], functions=[partial_max(value#2007L)], output=[key#2006, max#2023L])
E                                  +- Union
...
```

> Basically by the plan output here, it looks like this is an AQE optimization that is turning the entire plan into a `LocalTableScan`
> 
> 
> 
> ```
> 
> E                   py4j.protocol.Py4JJavaError: An error occurred while calling z:org.apache.spark.sql.rapids.ExecutionPlanCaptureCallback.assertContains.
> 
> E                   : java.lang.AssertionError: assertion failed: Could not find DynamicPruningExpression in the Spark plan
> 
> E                   AdaptiveSparkPlan isFinalPlan=true
> 
> E                   +- == Final Plan ==
> 
> E                      LocalTableScan <empty>, [key#2006, max(value)#2017L]
> 
> E                   +- == Initial Plan ==
> 
> E                   +- == Initial Plan ==
> 
> E                      Sort [key#2006 ASC NULLS FIRST, max(value)#2017L ASC NULLS FIRST], true, 0
> 
> E                      +- Exchange rangepartitioning(key#2006 ASC NULLS FIRST, max(value)#2017L ASC NULLS FIRST, 4), ENSURE_REQUIREMENTS, [plan_id=5302]
> 
> E                         +- HashAggregate(keys=[key#2006], functions=[max(value#2007L)], output=[key#2006, max(value)#2017L])
> 
> E                            +- Exchange hashpartitioning(key#2006, 4), ENSURE_REQUIREMENTS, [plan_id=5299]
> 
> E                               +- HashAggregate(keys=[key#2006], functions=[partial_max(value#2007L)], output=[key#2006, max#2023L])
> 
> E                                  +- Union
> 
> ...
> 
> ```



I guess it determined via the join that this would return empty

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -330,18 +330,26 @@ class UniqueLongGen(DataGen):
 
 class RepeatSeqGen(DataGen):
     """Generate Repeated seq of `length` random items"""
-    def __init__(self, child, length):
-        super().__init__(child.data_type, nullable=False)
-        self.nullable = child.nullable
+    def __init__(self, child, length=None, data_type=None, nullable=False):
+        if isinstance(child, list):
+            super().__init__(data_type, nullable=False)
+            self.nullable = nullable
+            self._length = len(child)
+        else:
+            super().__init__(child.data_type, nullable=False)
+            self.nullable = child.nullable
+            assert(length is not None)
+            self._length = length
         self._child = child
         self._vals = []
-        self._length = length
         self._index = 0
 
     def __repr__(self):
         return super().__repr__() + '(' + str(self._child) + ')'
 
     def _cache_repr(self):
+        if isinstance(self._child, list):
+            return super()._cache_repr() + '(' + str(self._child) + ',' + str(self._length) + ')'
         return super()._cache_repr() + '(' + self._child._cache_repr() + ',' + str(self._length) + ')'
 
     def _loop_values(self):
@@ -351,9 +359,12 @@ class RepeatSeqGen(DataGen):
 
     def start(self, rand):
         self._index = 0
-        self._child.start(rand)
         self._start(rand, self._loop_values)
-        self._vals = [self._child.gen() for _ in range(0, self._length)]
+        if isinstance(self._child, list):
+            self._vals = self._child
+        else:
+            self._child.start(rand)
+            self._vals = [self._child.gen() for _ in range(0, self._length)]
 
 class SetValuesGen(DataGen):
     """A set of values that are randomly selected"""

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -347,13 +347,14 @@ class RepeatSeqGen(DataGen):
             self.nullable = nullable
             assert (length is None or length < len(child))
             self._length = length if length is not None else len(child)
+            self._child = child[:length] if length is not None else child
         else:
             super().__init__(child.data_type, nullable=False)
             self.nullable = child.nullable
             assert(data_type is None or data_type != child.data_type)
             assert(length is not None)
             self._length = length
-        self._child = child
+            self._child = child
         self._vals = []
         self._index = 0
 

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -335,16 +335,14 @@ class RepeatSeqGen(DataGen):
     When child is a list:
         data_type must be specified
         length must be <= length of child
-        nullable is honored
     When child is a DataGen:
         length must be specified
         data_type must be None or match child's
-        nullable is set child's nullable attribute
     """
-    def __init__(self, child, length=None, data_type=None, nullable=False):
+    def __init__(self, child, length=None, data_type=None):
         if isinstance(child, list):
             super().__init__(data_type, nullable=False)
-            self.nullable = nullable
+            self.nullable = None in child
             assert (length is None or length < len(child))
             self._length = length if length is not None else len(child)
             self._child = child[:length] if length is not None else child

--- a/integration_tests/src/main/python/data_gen.py
+++ b/integration_tests/src/main/python/data_gen.py
@@ -329,15 +329,28 @@ class UniqueLongGen(DataGen):
         self._start(rand, lambda: self.next_val())
 
 class RepeatSeqGen(DataGen):
-    """Generate Repeated seq of `length` random items"""
+    """Generate Repeated seq of `length` random items if child is a DataGen,
+    otherwise repeat the provided seq when child is a list.
+
+    When child is a list:
+        data_type must be specified
+        length must be <= length of child
+        nullable is honored
+    When child is a DataGen:
+        length must be specified
+        data_type must be None or match child's
+        nullable is set child's nullable attribute
+    """
     def __init__(self, child, length=None, data_type=None, nullable=False):
         if isinstance(child, list):
             super().__init__(data_type, nullable=False)
             self.nullable = nullable
-            self._length = len(child)
+            assert (length is None or length < len(child))
+            self._length = length if length is not None else len(child)
         else:
             super().__init__(child.data_type, nullable=False)
             self.nullable = child.nullable
+            assert(data_type is None or data_type != child.data_type)
             assert(length is not None)
             self._length = length
         self._child = child

--- a/integration_tests/src/main/python/dpp_test.py
+++ b/integration_tests/src/main/python/dpp_test.py
@@ -20,9 +20,9 @@ from data_gen import *
 from marks import ignore_order, allow_non_gpu
 from spark_session import is_before_spark_320, with_cpu_session, is_before_spark_312, is_databricks_runtime, is_databricks113_or_later
 
-# non-positive values here can produce a degenerative join, so we want a filter value associated
-# with only positive values. See https://github.com/NVIDIA/spark-rapids/issues/10147
-value_gen = RepeatSeqGen(int_gen, length=100) 
+# non-positive values here can produce a degenerative join, so here we ensure that most values are
+# positive to ensure the join will produce rows. See https://github.com/NVIDIA/spark-rapids/issues/10147
+value_gen = RepeatSeqGen(IntegerGen(nullable=True, min_val=1, max_val=100, special_cases=[0,-1]), length=100)
 
 def create_dim_table(table_name, table_format, length=500):
     def fn(spark):

--- a/integration_tests/src/main/python/dpp_test.py
+++ b/integration_tests/src/main/python/dpp_test.py
@@ -27,7 +27,9 @@ def create_dim_table(table_name, table_format, length=500):
             ('key', IntegerGen(nullable=False, min_val=0, max_val=9, special_cases=[])),
             ('skey', IntegerGen(nullable=False, min_val=0, max_val=4, special_cases=[])),
             ('ex_key', IntegerGen(nullable=False, min_val=0, max_val=3, special_cases=[])),
-            ('value', int_gen),
+            # non-positive values here can produce a degenerative join, so we want a filter value associated
+            # with only positive values. See https://github.com/NVIDIA/spark-rapids/issues/10147
+            ('value', IntegerGen(min_val=1, special_cases=[1, INT_MAX])),
             # specify nullable=False for `filter` to avoid generating invalid SQL with
             # expression `filter = None` (https://github.com/NVIDIA/spark-rapids/issues/9817)
             ('filter', RepeatSeqGen(

--- a/integration_tests/src/main/python/dpp_test.py
+++ b/integration_tests/src/main/python/dpp_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2021-2023, NVIDIA CORPORATION.
+# Copyright (c) 2021-2024, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -20,6 +20,9 @@ from data_gen import *
 from marks import ignore_order, allow_non_gpu
 from spark_session import is_before_spark_320, with_cpu_session, is_before_spark_312, is_databricks_runtime, is_databricks113_or_later
 
+# non-positive values here can produce a degenerative join, so we want a filter value associated
+# with only positive values. See https://github.com/NVIDIA/spark-rapids/issues/10147
+value_gen = RepeatSeqGen(int_gen, length=100) 
 
 def create_dim_table(table_name, table_format, length=500):
     def fn(spark):
@@ -27,9 +30,7 @@ def create_dim_table(table_name, table_format, length=500):
             ('key', IntegerGen(nullable=False, min_val=0, max_val=9, special_cases=[])),
             ('skey', IntegerGen(nullable=False, min_val=0, max_val=4, special_cases=[])),
             ('ex_key', IntegerGen(nullable=False, min_val=0, max_val=3, special_cases=[])),
-            # non-positive values here can produce a degenerative join, so we want a filter value associated
-            # with only positive values. See https://github.com/NVIDIA/spark-rapids/issues/10147
-            ('value', IntegerGen(min_val=1, special_cases=[1, INT_MAX])),
+            ('value', value_gen),
             # specify nullable=False for `filter` to avoid generating invalid SQL with
             # expression `filter = None` (https://github.com/NVIDIA/spark-rapids/issues/9817)
             ('filter', RepeatSeqGen(
@@ -51,7 +52,7 @@ def create_fact_table(table_name, table_format, length=2000):
             ('skey', IntegerGen(nullable=False, min_val=0, max_val=4, special_cases=[])),
             # ex_key is not a partition column
             ('ex_key', IntegerGen(nullable=False, min_val=0, max_val=3, special_cases=[])),
-            ('value', int_gen)], length)
+            ('value', value_gen)], length)
         df.write.format(table_format) \
             .mode("overwrite") \
             .partitionBy('key', 'skey') \

--- a/integration_tests/src/main/python/dpp_test.py
+++ b/integration_tests/src/main/python/dpp_test.py
@@ -14,6 +14,8 @@
 
 import pytest
 
+from pyspark.sql.types import IntegerType
+
 from asserts import assert_cpu_and_gpu_are_equal_collect_with_capture, assert_gpu_and_cpu_are_equal_collect
 from conftest import spark_tmp_table_factory
 from data_gen import *
@@ -22,7 +24,7 @@ from spark_session import is_before_spark_320, with_cpu_session, is_before_spark
 
 # non-positive values here can produce a degenerative join, so here we ensure that most values are
 # positive to ensure the join will produce rows. See https://github.com/NVIDIA/spark-rapids/issues/10147
-value_gen = RepeatSeqGen(IntegerGen(nullable=True, min_val=1, max_val=100, special_cases=[0,-1]), length=100)
+value_gen = RepeatSeqGen([None, INT_MIN, -1, 0, 1, INT_MAX], data_type=IntegerType(), nullable=True)
 
 def create_dim_table(table_name, table_format, length=500):
     def fn(spark):

--- a/integration_tests/src/main/python/dpp_test.py
+++ b/integration_tests/src/main/python/dpp_test.py
@@ -24,7 +24,7 @@ from spark_session import is_before_spark_320, with_cpu_session, is_before_spark
 
 # non-positive values here can produce a degenerative join, so here we ensure that most values are
 # positive to ensure the join will produce rows. See https://github.com/NVIDIA/spark-rapids/issues/10147
-value_gen = RepeatSeqGen([None, INT_MIN, -1, 0, 1, INT_MAX], data_type=IntegerType(), nullable=False)
+value_gen = RepeatSeqGen([None, INT_MIN, -1, 0, 1, INT_MAX], data_type=IntegerType())
 
 def create_dim_table(table_name, table_format, length=500):
     def fn(spark):

--- a/integration_tests/src/main/python/dpp_test.py
+++ b/integration_tests/src/main/python/dpp_test.py
@@ -24,7 +24,7 @@ from spark_session import is_before_spark_320, with_cpu_session, is_before_spark
 
 # non-positive values here can produce a degenerative join, so here we ensure that most values are
 # positive to ensure the join will produce rows. See https://github.com/NVIDIA/spark-rapids/issues/10147
-value_gen = RepeatSeqGen([None, INT_MIN, -1, 0, 1, INT_MAX], data_type=IntegerType(), nullable=True)
+value_gen = RepeatSeqGen([None, INT_MIN, -1, 0, 1, INT_MAX], data_type=IntegerType(), nullable=False)
 
 def create_dim_table(table_name, table_format, length=500):
     def fn(spark):

